frontend/packages/console-dynamic-plugin-sdk/src/extensions/global-configs.ts

```diff
@@ -0,0 +1,29 @@
+import { Extension } from '@console/plugin-sdk/src/typings/base';
+
+export namespace ExtensionProperties {
+  export type GlobalConfig = {
```

Comment:
This file could be named `cluster-settings.ts` and its extension & properties types could be named `ClusterGlobalConfig` to better reflect the fact that it's meant to extend core components under `public/components/cluster-settings`.

I also suggest grouping related bits of data together, while also reducing the amount of data to a necessary minimum. For example, referring to a k8s resource strictly via `group/version/kind` instead of using the wrapper `K8sKind` type.

```ts
/** The model which refers to a cluster config resource. */
model: {
  group: string;
  version: string;
  kind: string;
}
```